src/__tests__/components/fields/text-field/text-field.spec.tsx

```diff
@@ -105,9 +107,60 @@ describe(DEFAULT_FIELD_TYPE, () => {
 		expect(getTextfield()).toHaveAttribute("readonly");
 		expect(getTextfield()).toBeDisabled();
 	});
+
+	it("should not prevent copy and paste", async () => {
+		renderComponent({
+			customOptions: {
+				preventCopyAndPaste: false,
+			},
+		});
+		const textField = getTextfield();
+		textField.focus();
+		await act(async () => {
+			await userEvent.paste(EXPECTED_TEXT);
+			expect(textField).toHaveValue(EXPECTED_TEXT);
+		});
+	});
+
+	it("should prevent copy and paste", async () => {
+		renderComponent({
+			customOptions: {
+				preventCopyAndPaste: true,
+			},
+		});
+		const textField = getTextfield();
+		textField.focus();
+		await act(async () => {
+			await userEvent.paste(EXPECTED_TEXT);
+			expect(textField).not.toHaveValue(EXPECTED_TEXT);
+		});
+	});
+
+	it("should fireEvent return false since default behavior is prevented", () => {
```

Comment:
- `should prevent drag & drop text into field if preventDragAndDrop is true`
- add a comment before this test to explain why the test is written this way
`// testing the return value of drop event because @testing-library/user-event does not respect preventDefault()`